### PR TITLE
chore(knob) add option to set min, max and step on number knob

### PR DESCRIPTION
**Pull request by assuncaocharles**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Include a change request file using `$ yarn change`

#### Description of changes

 Add option to set min, max and step on number knob

#### Focus areas to test

(optional)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/fluentui/pull/12653)

### CODE REVIEW

File: packages/fluentui/docs-components/src/knobs/defaultComponents.tsx [hunk 1]
```diff
@@ -35,11 +41,16 @@ const KnobBoolean: React.FunctionComponent<KnobComponentProps> = props => (
...
 );
 
-const KnobNumber: React.FunctionComponent<KnobComponentProps> = props => (
+const KnobNumber: React.FunctionComponent<KnobNumberKnobComponentProps> = props => (
   <input
     onChange={(e: React.ChangeEvent<HTMLInputElement>) => {
-      props.setValue(parseInt(e.target.value, 10));
+      let newValue = parseInt(e.target.value, 10);
+      newValue = newValue < props.min ? props.min : newValue > props.max ? props.max : newValue;
+      props.setValue(newValue);
```
Comment (layershifter) [inline packages/fluentui/docs-components/src/knobs/defaultComponents.tsx:49]:
```suggestion
      props.setValue(parseInt(e.target.value, 10));
```

This looks redundant as `input` with `type=number` already handles this: https://www.w3schools.com/tags/tryit.asp?filename=tryhtml5_input_type_number

File: packages/fluentui/docs-components/src/knobs/types.ts [hunk 2]
```diff
@@ -40,6 +40,12 @@ export type KnobRangeKnobComponentProps = KnobComponentProps & {
   unit: string;
 };
 
+export type KnobNumberKnobComponentProps = KnobComponentProps & {
+  min: number;
+  max: number;
+  step: number;
```
Comment (layershifter) [inline packages/fluentui/docs-components/src/knobs/types.ts:46]:
```suggestion
  min: string;
  max: string;
  step: string;
```

Just to have the same as in `KnobRangeKnobComponentProps`

File: packages/fluentui/docs-components/src/knobs/useNumberKnob.ts [hunk 3]
```diff
@@ -1,11 +1,26 @@
...
+type UseNumberKnobSpecificOptions = {
+  min?: number;
+  max?: number;
+  step?: number;
+};
+
+type UseNumberKnobOptions<T> = UseKnobOptions<T> & UseNumberKnobSpecificOptions;
+
+const useStringKnob = (options: UseNumberKnobOptions<number>) => {
+  const { initialValue = 0, min = 0, max = parseValue(initialValue), step = 1, ...rest } = options;
```
Comment (layershifter) [inline packages/fluentui/docs-components/src/knobs/useNumberKnob.ts:14]:
```suggestion
  const { initialValue = 0, min = 0, max = undefined, step = 1, ...rest } = options;
```

Should it be just `undefined` to keep existing behavior?

File: packages/fluentui/docs-components/src/knobs/defaultComponents.tsx (hunk 1 above)
Comment (assuncaocharles) [reply packages/fluentui/docs-components/src/knobs/defaultComponents.tsx:49]: It doesn't if the user actually type a bigger number than the max...

File: packages/fluentui/docs-components/src/knobs/types.ts (hunk 2 above)
Comment (assuncaocharles) [reply packages/fluentui/docs-components/src/knobs/types.ts:46]: Done

File: packages/fluentui/docs-components/src/knobs/useNumberKnob.ts (hunk 3 above)
Comment (assuncaocharles) [reply packages/fluentui/docs-components/src/knobs/useNumberKnob.ts:14]: Done

**Pull request merged**